scripts/score.py

```diff
@@ -0,0 +1,138 @@
+    baseline_avg_by_series: dict[str, dict[tuple[str, str], float]] = {}
+    summary: list[str] = []
+
+    for series in series_list:
+        series_cfg = (baselines_cfg.get("series", {}) or {}).get(series, {}) if isinstance(baselines_cfg, dict) else {}
+        base_provider = series_cfg.get("provider") or (baselines_cfg.get("default", {}) or {}).get("provider")
+        base_device = series_cfg.get("device") or (baselines_cfg.get("default", {}) or {}).get("device")
+
+        if not base_provider or not base_device:
+            baseline_avg_by_series[series] = {}
+            summary.append(f"{series}: (no baseline configured)")
+            continue
+
+        selected = [
+            r for r in flat_rows
+            if r.get("provider") == base_provider and r.get("device") == base_device and row_series.get(id(r)) == series
+        ]
+
+        baseline_values: dict[tuple[str, str], list[float]] = {}
+        for r in selected:
+            results = r.get("results") if isinstance(r.get("results"), dict) else {}
+            bench = derive_benchmark_name(r)
+            for metric, val in results.items():
+                try:
+                    num = float(val)
+                except Exception:
+                    continue
+                if not (num == num):  # NaN
+                    continue
+                baseline_values.setdefault((bench, metric), []).append(num)
+
+        baseline_avg: dict[tuple[str, str], float] = {}
+        for key, vals in baseline_values.items():
+            if not vals:
+                continue
+            try:
+                baseline_avg[key] = sum(vals) / len(vals)
+            except Exception:
+                pass
+        baseline_avg_by_series[series] = baseline_avg
+        summary.append(f"{series}: {base_provider}/{base_device}")
+
+    return baseline_avg_by_series, summary
+
+
+def compute_and_attach_metriq_scores(
+    flat_rows: list[dict[str, Any]],
+    row_series: dict[int, str],
+    baseline_avg_by_series: dict[str, dict[tuple[str, str], float]],
+) -> None:
+    for r in flat_rows:
+        results = r.get("results") if isinstance(r.get("results"), dict) else {}
+        if not results:
+            continue
+        bench = derive_benchmark_name(r)
+        dir_map = r.get("directions") if isinstance(r.get("directions"), dict) else {}
+        series = row_series.get(id(r))
+        baseline_avg = baseline_avg_by_series.get(series or "", {})
+        scores: dict[str, float] = {}
+        for metric, val in results.items():
+            try:
+                v = float(val)
+            except Exception:
+                continue
+            if not (v == v):  # NaN
+                continue
+            base = baseline_avg.get((bench, metric))
+            if base is None:
+                continue
+            direction = str(dir_map.get(metric, "higher")).lower()
+            score: Optional[float] = None
+            try:
```

Comment:
There's a lot of try/excepts like this that likely can be consolidated/removed/tidied.